app/main.py

```diff
@@ -1,34 +1,134 @@
+        self.is_drowned = is_drowned
         # Create decks and save them to a list `self.decks`
-        pass
+        self.decks: list = []
+        for row in range(start[0], end[0] + 1):
+            for column in range(start[1], end[1] + 1):
+                self.decks.append(Deck(row, column))
 
-    def get_deck(self, row, column):
+    def get_deck(self, row: int, column: int) -> Deck:
         # Find the corresponding deck in the list
-        pass
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
 
-    def fire(self, row, column):
+    def fire(self, row: int, column: int) -> None:
         # Change the `is_alive` status of the deck
+        deck_fire = self.get_deck(row, column)
+        deck_fire.is_alive = False
+
         # And update the `is_drowned` value if it's needed
-        pass
+        is_alive: bool = False
+        for deck in self.decks:
+            if deck.is_alive:
+                is_alive = True
+                break
+        if not is_alive:
+            self.is_drowned = True
 
 
 class Battleship:
-    def __init__(self, ships):
+    def __init__(self, ships: list) -> None:
         # Create a dict `self.field`.
         # Its keys are tuples - the coordinates of the non-empty cells,
         # A value for each cell is a reference to the ship
         # which is located in it
-        pass
+        self.ships: list = []
+        self.field: dict = {}
+        for ship_coord in ships:
+            ship = Ship(*ship_coord)
+            self.ships.append(ship)
+            for deck in ship.decks:
+                self.field[(deck.row, deck.column)] = ship
+        try:
+            self._validate_field()
+        except ValueError as verr:
+            print(verr)
+            exit()
 
-    def fire(self, location: tuple):
+    def fire(self, location: tuple) -> str:
         # This function should check whether the location
         # is a key in the `self.field`
         # If it is, then it should check if this cell is the last alive
         # in the ship or not.
-        pass
+        if location in self.field.keys():
+            ship = self.field[location]
+            ship.fire(*location)
+            if ship.is_drowned:
+                return "Sunk!"
+            if not ship.get_deck(*location).is_alive:
+                return "Hit!"
+        return "Miss!"
+
+    def print_field(self) -> None:
+        end_value: str = "     "
+        for row in range(10):
+            for col in range(10):
+                if (row, col) not in self.field.keys():
+                    print("~", end=end_value)
+                else:
+                    for coord, ship in self.field.items():
+                        if (row, col) == coord:
+                            if ship.is_drowned:
+                                print("x", end=end_value)
+                            else:
+                                for deck in ship.decks:
```

Comment:
`Flat is better than nested.` due to the zen of python

this code seems too nested. try to fix it